RecoTracker/LSTCore/interface/LSTESData.h

```diff
@@ -0,0 +1,96 @@
+#ifndef RecoTracker_LSTCore_interface_LSTESData_h
+#define RecoTracker_LSTCore_interface_LSTESData_h
+
+#include "RecoTracker/LSTCore/interface/Common.h"
+#include "RecoTracker/LSTCore/interface/EndcapGeometryDevHostCollection.h"
+#include "RecoTracker/LSTCore/interface/ModulesHostCollection.h"
+#include "RecoTracker/LSTCore/interface/PixelMap.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToDevice.h"
+
+#include <filesystem>
+#include <memory>
+
+namespace lst {
+
+  template <typename TDev>
+  struct LSTESData {
+    uint16_t nModules;
+    uint16_t nLowerModules;
+    unsigned int nPixels;
+    unsigned int nEndCapMap;
+    std::unique_ptr<const PortableMultiCollection<TDev, ModulesSoA, ModulesPixelSoA>> modules;
+    std::unique_ptr<const PortableCollection<EndcapGeometryDevSoA, TDev>> endcapGeometry;
+    // Host-side object that is shared between the LSTESData<TDev> objects for different devices
+    std::shared_ptr<const PixelMap> pixelMapping;
+
+    LSTESData(uint16_t const& nModulesIn,
+              uint16_t const& nLowerModulesIn,
+              unsigned int const& nPixelsIn,
+              unsigned int const& nEndCapMapIn,
+              std::unique_ptr<const PortableMultiCollection<TDev, ModulesSoA, ModulesPixelSoA>> modulesIn,
+              std::unique_ptr<const PortableCollection<EndcapGeometryDevSoA, TDev>> endcapGeometryIn,
+              std::shared_ptr<const PixelMap> const& pixelMappingIn)
+        : nModules(nModulesIn),
+          nLowerModules(nLowerModulesIn),
+          nPixels(nPixelsIn),
+          nEndCapMap(nEndCapMapIn),
+          modules(std::move(modulesIn)),
+          endcapGeometry(std::move(endcapGeometryIn)),
+          pixelMapping(pixelMappingIn) {}
+  };
+
+  std::unique_ptr<LSTESData<alpaka_common::DevHost>> loadAndFillESHost();
+
+}  // namespace lst
+
+namespace cms::alpakatools {
+
+  // The templated definition in CMSSW doesn't work when using CPU as the device
+  template <>
+  struct CopyToDevice<PortableHostCollection<lst::EndcapGeometryDevSoA>> {
+    template <typename TQueue>
```

Comment:
Hmm yeah, we can use some `if constexpr` in `CopyToDevice<lst::LSTESData<alpaka_common::DevHost>>` now that we figured out how to get it to compile.

Reply:
I'll do this once #46636 is merged

Reply:
@makortel 
the fix is available in https://github.com/SegmentLinking/cmssw/pull/120,
but it relies on  #46636.

It looks like #46636 still fails some tests and is not ready.
Does it make sense to 
- a) merge the update from https://github.com/SegmentLinking/cmssw/pull/120 and aim to "cmsbot please test with #46636"  (this will break our CI or require to on-the-fly merge of #46636)
- b) or should that update in LST code be added later (possibly without being the change blocking merge of this PR)?  IIUC, the update is needed for more safe/stricter template implementation, (that the current code is operational).

Please advise.

Reply:
#46636 was merged yesterday. If you updated this PR with https://github.com/SegmentLinking/cmssw/pull/120, I think we can call it a day for this PR, and address follow-ups in subsequent PRs.